models/cable/R/write.config.CABLE.R

```diff
@@ -0,0 +1,125 @@
+  
+  # find out where to write run/ouput
+  rundir <- file.path(settings$host$rundir, run.id)
+  outdir <- file.path(settings$host$outdir, run.id)
+  
+  #-----------------------------------------------------------------------
+  # create launch script (which will create symlink)
+  if (!is.null(settings$model$jobtemplate) && file.exists(settings$model$jobtemplate)) {
+    jobsh <- readLines(con = settings$model$jobtemplate, n = -1)
+  } else {
+    jobsh <- readLines(con = system.file("template.job", package = "PEcAn.CABLE"), n = -1)
+  }
+  
+  # create host specific setttings
+  hostsetup <- ""
+  if (!is.null(settings$model$prerun)) {
+    hostsetup <- paste(hostsetup, sep = "\n", paste(settings$model$prerun, collapse = "\n"))
+  }
+  if (!is.null(settings$host$prerun)) {
+    hostsetup <- paste(hostsetup, sep = "\n", paste(settings$host$prerun, collapse = "\n"))
+  }
+  
+  hostteardown <- ""
+  if (!is.null(settings$model$postrun)) {
+    hostteardown <- paste(hostteardown, sep = "\n", paste(settings$model$postrun, collapse = "\n"))
+  }
+  if (!is.null(settings$host$postrun)) {
+    hostteardown <- paste(hostteardown, sep = "\n", paste(settings$host$postrun, collapse = "\n"))
+  }
+  
+  # create job.sh
+  jobsh <- gsub("@HOST_SETUP@", hostsetup, jobsh)
+  jobsh <- gsub("@HOST_TEARDOWN@", hostteardown, jobsh)
+  
+  jobsh <- gsub("@SITE_LAT@", settings$run$site$lat, jobsh)
+  jobsh <- gsub("@SITE_LON@", settings$run$site$lon, jobsh)
+  jobsh <- gsub("@SITE_MET@", settings$run$site$met, jobsh)
+  
+  jobsh <- gsub("@START_DATE@", settings$run$start.date, jobsh)
+  jobsh <- gsub("@END_DATE@", settings$run$end.date, jobsh)
+  
+  jobsh <- gsub("@OUTDIR@", outdir, jobsh)
+  jobsh <- gsub("@RUNDIR@", rundir, jobsh)
+  
+  jobsh <- gsub("@BINARY@", settings$model$binary, jobsh)
+  
+  writeLines(jobsh, con = file.path(settings$rundir, run.id, "job.sh"))
+  Sys.chmod(file.path(settings$rundir, run.id, "job.sh"))
+  
+  #-----------------------------------------------------------------------
+  ### Edit a templated config file for runs
+  if (!is.null(settings$model$config) && file.exists(settings$model$config)) {
+    config.text <- readLines(con = settings$model$config, n = -1)
+  } else {
+    filename <- system.file("cable.nml", package = "PEcAn.CABLE")
+    if (filename == "") {
+      if (!is.null(settings$model$revision)) {
+        filename <- system.file(paste0("config.", settings$model$revision), package = "PEcAn.CABLE")
+      } else {
+        model <- db.query(paste("SELECT * FROM models WHERE id =", settings$model$id), params = settings$database$bety)
+        filename <- system.file(paste0("config.r", model$revision), package = "PEcAn.CABLE")
+      }
+    }
+    if (filename == "") {
+      logger.severe("Could not find config template")
+    }
+    logger.info("Using", filename, "as template")
+    config.text <- readLines(con = filename, n = -1)
+  }
+  
+  config.text <- gsub("@SITE_LAT@", settings$run$site$lat, config.text)
```

Comment:
@kragosta for future reference, once we get CABLE to run, an important next step is to start putting these @ tags into the template cable.nml you added (especially the lines that say ```*** SET PATH IN cable.nml ***```)  so that PEcAn can start to find/replace the tags with the correct settings on the fly.  That may require adding additional tags, as well as realizing that some the tags here are not needed or need to go into different files than the main cable.nml